TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -1223,14 +1223,34 @@ public static void CheckSection(Section section)
 
         public static void CheckParagraph(Paragraph paragraph)
         {
-            //Get all paragraphs with the same name and having the same section name
-            var paragraphs = paragraph.SymbolTable.GetParagraphs(p => p.Name.Equals(paragraph.Name, StringComparison.OrdinalIgnoreCase) && p.SemanticData.Owner == paragraph.SemanticData.Owner).ToList();
-
-            //Paragraphs can't have the same name within the same section
-            if (paragraphs.Count > 1)
+            Scopes.Domain<ParagraphSymbol> paragraphs = null;
+            // Get the Owner's Paragraph Domain
+            switch (paragraph.SemanticData.Owner.Kind)
+            {
+                case Symbol.Kinds.Program:
+                case Symbol.Kinds.Function:
+                    {
+                        ScopeSymbol owner = (ScopeSymbol)paragraph.SemanticData.Owner;
+                        paragraphs = owner.Paragraphs;
+                    }
+                    break;
+                case Symbol.Kinds.Section:
+                    {
+                        SectionSymbol section = (SectionSymbol)paragraph.SemanticData.Owner;
+                        paragraphs = section.Paragraphs;
+                    }
+                    break;
+                default:
+                    System.Diagnostics.Debug.Fail("A Program, a Function or a Section was expected as paragraph's owner.");
+                    break;
+            }
+            // Look for the paragraph in the domain
+            Scopes.Container<ParagraphSymbol>.Entry entry = paragraphs.Lookup(paragraph.Name);
```

Comment:
The only way to resolve a Symbol is the SymbolTable, we've already discuss about that.